infra/graph_generators_test.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar 20 17:17:54 2019
+
+@author: Zargham
+"""
+import networkx as nx
+import unittest
+from graph_generators import lineGraphGen, starGraphGen, treeGraphGen
+
+class GraphGenerators_LineGraph(unittest.TestCase):
+    def test_lineGraphGen_sanity(self):
+        g = lineGraphGen(4)
+        self.assertEqual(list(g.edges()), [(0, 1), (1, 2), (2, 3)])
+
+    def test_lineGraphGen_bidir(self):
+        g = lineGraphGen(4, bidir=True)
+        # (node, neighbor)
+        self.assertEqual(list(g.edges()), [(0, 1), (1, 0), (1, 2), (2, 1), (2, 3), (3, 2)])
+
+class GraphGenerators_StarGraph(unittest.TestCase):
+
+# how to validate that 'kind' is either 'source', 'sink', or 'bidir' ?? 
+
+    def test_starGraph_sink(self):
+        g = starGraphGen(4)
+        self.assertEqual(list(g.edges()), [(1, 0), (2, 0), (3, 0), (4, 0)])
+
+    def test_starGraph_source(self):
+        g = starGraphGen(4, kind='source')
+        self.assertEqual(list(g.edges()), [(0, 1), (0, 2), (0, 3), (0, 4)])
+
+    def test_starGraph_bidir(self):
+        g = starGraphGen(4, kind='bidir')
+        self.assertEqual(list(g.edges()), [(0, 1), (0, 2), (0, 3), (0, 4), (1, 0), (2, 0), (3, 0), (4, 0)])
```

Comment:
this is kind of a doozy to keep all in my head--maybe check for `n=2` or `n=1` instead.

Reply:
I kept them at mostly 3 and 4 because for 2 node graphs the there would be no way to distinguish some of the graph patterns; for example the star and the line would be the same.